plugins/kernel_v2/config/disk.rb

```diff
@@ -164,6 +165,10 @@ def validate(machine)
           end
         end
 
+        if @type == :dvd && !@file
+          errors << I18n.t("vagrant.config.disk.dvd_type_file_required", name: @name, machine: machine.name)
+        end
```

Comment:
I think we also need a validation step to check if type is DVD but has been set to `primary`. Since `primary` is intended for hard disks rather than DVDs.